alternator/executor.cc

```diff
@@ -3033,9 +3033,9 @@ struct primary_key_equal {
 // mutations to the same partition.
 class put_or_delete_item_cas_request : public service::cas_request {
     schema_ptr schema;
-    std::vector<put_or_delete_item> _mutation_builders;
+    const std::vector<put_or_delete_item>& _mutation_builders;
```

Comment:
This patch is scrary, there's a lot that can go wrong here. Instead sending the free() command to the original shard, we now need to rely on the original shard remembering to do the free() at the right time. It's not that it can't be made to work, it's just that there are more ways it can be terribly wrong like classic C code with pointers.